### PR TITLE
permutation program

### PR DESCRIPTION
**Pull request by joynewton1996**

### CODE REVIEW

File: hacker_Rank/May11_Answer.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+def factorial_list(n):
+    factorial = []
+    step = 0
+    while step < n and n > 1:
+        k = (n - step)
+        step += 1
+        factorial.append(k)
+
```
Comment (roy-einstein) [inline hacker_Rank/May11_Answer.py]: This can be simplified with single loop,

**Review by roy-einstein — CHANGES_REQUESTED** [review]
Rejected the logic is in-correct

**Review by roy-einstein — CHANGES_REQUESTED** [review]
Solution would be,
Find Max number,
generate series with Max Number, 1....N,
then start looking for numbers present in given array with Series Generated

File: hacker_Rank/May11_Answer.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    if n == 1:
+        return factorial[1]
+
+    return factorial
```
Comment (roy-einstein) [inline hacker_Rank/May11_Answer.py]: Why do you need this ?

File: hacker_Rank/May11_Answer.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from algorithm import Algo
+
+obj = Algo()
+
+
+def solutuion(a):
+    print(a)
+    max_value = obj.max_value(a)
+    factorial = factorial_list(max_value)
```
Comment (roy-einstein) [inline hacker_Rank/May11_Answer.py]: ``` factorial = range(1,max_value+1) ``` 
 ???

**Review by roy-einstein — CHANGES_REQUESTED** [review]
Learn and USE ``` sets ``` 
``` difference, intersection ``` 
which can be easily solved

File: hacker_Rank/May11_Answer.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+from algorithm import Algo
+
+obj = Algo()
+
+
+def solutuion(a):
+    max_value = obj.max_value(a)
+    series = set(item for item in range(1, max_value + 1))
+    a = set(a)
+    difference = series.difference(a)
```
Comment (roy-einstein) [inline hacker_Rank/May11_Answer.py]: ``` a = set(a) ``` <-- is this required?, did you try without using?
``` series.difference(a) ``` will not work ?

**Review by roy-einstein — CHANGES_REQUESTED** [review]
need litle more work

File: hacker_Rank/May11_Answer.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+from algorithm import Algo
+
+obj = Algo()
+
+
+def solutuion(a):
+    max_value = obj.max_value(a)
+    series = set(item for item in range(1, max_value + 1))
```
Comment (roy-einstein) [inline hacker_Rank/May11_Answer.py]: Can we write simplified way of generating of series ?
``` list(range(1,max_value+1)) ``` 
will not solve problem ?

**Review by roy-einstein — CHANGES_REQUESTED** [review]

**Review by roy-einstein — CHANGES_REQUESTED** [review]
Needs work

File: hacker_Rank/May11_Answer.py (hunk 4 above)
Comment (joynewton1996) [reply hacker_Rank/May11_Answer.py]: okay i will check

**Review by roy-einstein — APPROVED** [review]
keep good work

**Pull request merged**